pkg/controller/etcdproxy/etcdproxy_controller.go

```diff
@@ -435,3 +453,110 @@ func (c *EtcdProxyController) handleObject(obj interface{}) {
 		return
 	}
 }
+
+// createEtcdProxyCACertConfigMap creates ConfigMap in controller namespace with Etcd Proxy CA certificate.
+func (c EtcdProxyController) createEtcdProxyCACertConfigMap(etcdstorage *etcdstoragev1alpha1.EtcdStorage) error {
+	// ConfigMap in controller namespace for the etcd proxy CA certificate.
+	caConfigMap, err := c.kubeclientset.CoreV1().ConfigMaps(c.config.ControllerNamespace).
+		Get(etcdProxyCAConfigMapName(etcdstorage), metav1.GetOptions{})
+	if errors.IsNotFound(err) {
+		data := map[string]string{"ca.pem": EtcdProxyCACert}
+		caConfigMap, err = c.kubeclientset.CoreV1().ConfigMaps(c.config.ControllerNamespace).
+			Create(newConfigMap(etcdstorage, etcdProxyCAConfigMapName(etcdstorage), c.config.ControllerNamespace, data))
+	}
+	if err != nil {
+		c.recorder.Event(etcdstorage, corev1.EventTypeWarning, ErrUnknown, err.Error())
+		return err
+	}
+	if !metav1.IsControlledBy(caConfigMap, etcdstorage) {
+		caConfigMap.SetOwnerReferences([]metav1.OwnerReference{
+			*metav1.NewControllerRef(etcdstorage, etcdstoragev1alpha1.SchemeGroupVersion.WithKind("EtcdStorage")),
+		})
+		caConfigMap, err = c.kubeclientset.CoreV1().ConfigMaps(c.config.ControllerNamespace).Update(caConfigMap)
```

Comment:
don't we need to make sure the data is up to date (in case you rotated) even when the ownerref matches your expectations?

Reply:
Yes, we actually need to make sure the data in Secrets/ConfigMaps is up to date. Updated the PR to include additional checks before updating.